mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -490,10 +484,61 @@ void CGUseList::decrementDiscardableUses(CGUser &Uses) {
 }
 
 //===----------------------------------------------------------------------===//
-// InlinerBase
+// InlineHeuristic
+//===----------------------------------------------------------------------===//
+
+bool InlineHeuristic::shouldInline(ResolvedCall &ResolvedCall,
+                                   const CGUseList &Uses) const {
+  if (isRecursiveCall(ResolvedCall))
+    return false;
+
+  FunctionOpInterface Callee = getCalledFunction(ResolvedCall.Call);
+  NamedAttrList FnAttrs(Callee->getAttrDictionary());
+  Optional<NamedAttribute> PassThroughAttr = FnAttrs.getNamed("passthrough");
+
+  bool ShouldInline = false;
+  switch (InlineMode) {
+  case sycl::InlineMode::Aggressive:
+    llvm_unreachable("TODO");
+  case sycl::InlineMode::Simple:
+    // Inline a function if it has an attribute suggesting that inlining is
+    // desirable.
+    if (PassThroughAttr)
+      ShouldInline |= llvm::any_of(
```

Comment:
```suggestion
      ShouldInline = llvm::any_of(
```

Will always be false.